hs/SvgParsing/Builder.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE OverloadedStrings, FlexibleContexts, GADTs, ScopedTypeVariables #-}
+module SvgParsing.Builder where
+
+import SvgParsing.Types
+import SvgParsing.ParserUtil
+import Control.Monad.IO.Class  (liftIO)
+import Database.Persist
+import Database.Persist.Sqlite
+import Data.Char
+import Data.List.Split
+import Data.List
+import Database.JsonParser
+import Database.Tables
+
+-- | Determines the source and target nodes of the path.
+buildPath :: [Shape] -> [Shape] -> Path -> Int -> Path
+buildPath rects ellipses entity idCounter
+    | pathIsRegion entity =
+        Path ('p' : show idCounter)
+             coords
+             (pathFill entity)
+             (pathStroke entity)
+             (pathIsRegion entity)
+             ""
+             ""
+    | otherwise =
+        let start = head coords
+            end = last coords
+            sourceNode = getIntersectingShape start (rects ++ ellipses)
+            targetNode = getIntersectingShape end (rects ++ ellipses)
+            in Path ('p' : show idCounter)
+                    coords
+                    (pathFill entity)
+                    (pathStroke entity)
+                    (pathIsRegion entity)
+                    sourceNode
+                    targetNode
+    where coords = pathPoints entity
+
+-- | Builds a Rect from a database entry in the rects table.
+buildRect :: [Text] -> Shape -> Shape
+buildRect texts entity =
+    let rectTexts = filter (\x -> intersects
+                            (shapeWidth entity)
+                            (shapeHeight entity)
+                            (shapeCoord entity)
+                            9
```

Comment:
Ah, that, yes I believe it is. I thought you were mentioning the intersecting parts- I'm going to see what I can do about that.